services/src/workflows/vector_stream.rs

```diff
@@ -0,0 +1,315 @@
+use crate::{contexts::Context, error::Result};
+use actix::{
+    fut::wrap_future, Actor, ActorContext, ActorFutureExt, AsyncContext, SpawnHandle, StreamHandler,
+};
+use actix_http::ws::{CloseCode, CloseReason, Item};
+use actix_web_actors::ws;
+use futures::{stream::BoxStream, FutureExt, StreamExt, TryFutureExt, TryStreamExt};
+use geoengine_datatypes::{collections::FeatureCollectionIpc, primitives::VectorQueryRectangle};
+use geoengine_operators::{
+    call_on_generic_vector_processor,
+    engine::{QueryAbortTrigger, QueryContext, QueryProcessorExt, VectorOperator},
+};
+
+pub struct VectorWebsocketStreamHandler {
+    state: VectorWebsocketStreamHandlerState,
+    abort_handle: Option<QueryAbortTrigger>,
+}
+
+type ByteStream = BoxStream<'static, StreamResult>;
+type StreamResult = Result<Vec<u8>>;
+
+enum VectorWebsocketStreamHandlerState {
+    Closed,
+    Idle { stream: ByteStream },
+    Processing { _fut: SpawnHandle },
+}
+
+impl Default for VectorWebsocketStreamHandlerState {
+    fn default() -> Self {
+        Self::Closed
+    }
+}
+
+impl Actor for VectorWebsocketStreamHandler {
+    type Context = ws::WebsocketContext<Self>;
+}
+
+impl StreamHandler<Result<ws::Message, ws::ProtocolError>> for VectorWebsocketStreamHandler {
+    fn started(&mut self, _ctx: &mut Self::Context) {}
+
+    fn finished(&mut self, ctx: &mut Self::Context) {
+        ctx.stop();
+
+        self.abort_processing();
+    }
+
+    fn handle(&mut self, msg: Result<ws::Message, ws::ProtocolError>, ctx: &mut Self::Context) {
+        match msg {
+            Ok(ws::Message::Ping(msg)) => ctx.pong(&msg),
+            Ok(ws::Message::Text(text)) if &text == "NEXT" => self.next_tile(ctx),
```

Comment:
can we name it next_chunk or something like this?

Reply:
[45ac504](https://github.com/geo-engine/geoengine/pull/747/commits/45ac50450977748902ee17981cdfd5d7b0327592)